### PR TITLE
Refactor classes for find command and dates

### PR DESCRIPTION
**Pull request by sumhungyee**
- Edit date arguments within sample data to reflect changes (delete time)
- Renamed NameContainsKeywordsPredicate to PersonContainsKeywordsPredicate
- Simplify regex expressions in Date and Age
- Edit Postal constructor to abide by SLAP

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ DengueHotspotTracker Level 3 (AB3) is a **desktop app for managing contacts, opt
 
    * `list` : Lists all cases.
 
-   * `add n/John Tan p/543299 d/2023-02-13 19:52 a/20` : Adds a contact named `John Tan` to the Dengue Hotspot Tracker.
+   * `add n/John Tan p/543299 d/2023-02-13 a/20` : Adds a contact named `John Tan` to the Dengue Hotspot Tracker.
```
Comment (valerietanhx) [inline docs/UserGuide.md:30]: All instances of "contact" in the User Guide should be changed to "case"!

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 2]
```diff
@@ -16,6 +16,6 @@ Person -up-> "*" Tag
 
 Person *--> Name
 Person *--> Phone
```
Comment (valerietanhx) [inline docs/diagrams/BetterModelClassDiagram.puml:18]: We'll need to update this to Postal Code

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 3]
```diff
@@ -16,6 +16,6 @@ Person -up-> "*" Tag
...
-Person *--> Email
+Person *--> Date
 Person *--> Address
```
Comment (valerietanhx) [inline docs/diagrams/BetterModelClassDiagram.puml:20]: And update this to age as well :)

Comment (valerietanhx) [reply docs/diagrams/BetterModelClassDiagram.puml:20]: This applies to all the class diagrams!

File: src/main/java/seedu/dengue/model/util/SampleDataUtil.java [hunk 4]
```diff
@@ -19,23 +19,23 @@
...
+                new Age("111"),
                 getTagSet("friends")),
-            new Person(new Name("Bernice Yu"), new Postal("992727"), new Date("berniceyu@example.com"),
-                new Age("Blk 30 Lorong 3 Serangoon Gardens, #07-18"),
+            new Person(new Name("Bernice Yu"), new Postal("992727"), new Date("2000-11-11"),
+                new Age("11"),
                 getTagSet("colleagues", "friends")),
             new Person(new Name("Charlotte Oliveiro"), new Postal("932283"),
-                    new Date("charlotte@example.com"), new Age("Blk 11 Ang Mo Kio Street 74, #11-04"),
+                    new Date("2000-11-11 00:00"), new Age("0"),
```
Comment (valerietanhx) [inline src/main/java/seedu/dengue/model/util/SampleDataUtil.java:29]:
```suggestion
                    new Date("2000-11-11"), new Age("0"),
```

File: src/main/java/seedu/dengue/model/person/Date.java [hunk 5]
```diff
@@ -2,35 +2,28 @@
...
-            + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_PART_REGEX = ALPHANUMERIC_NO_UNDERSCORE
-            + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
-    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
-    public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@" + DOMAIN_REGEX;
+            + "1. Date to be formatted as yyyy-MM-dd in 3 blocks separated by a hyphen '-'\n"
+            + " - Each block should only contain digits, and no other special characters\n"
+            + "2. Dates must be valid. i.e.\n"
+            + " - MM cannot take values >12";
```
Comment (valerietanhx) [inline src/main/java/seedu/dengue/model/person/Date.java:22]:
```suggestion
            + " - MM cannot take values <0 and >12"
            + " - dd cannot take values <0 and >31";
```

File: src/main/java/seedu/dengue/model/person/Age.java [hunk 6]
```diff
@@ -9,30 +9,30 @@
  */
 public class Age {
 
-    public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Ages must be a number not more than 3 digits long"
```
Comment (valerietanhx) [inline src/main/java/seedu/dengue/model/person/Age.java:12]:
```suggestion
    public static final String MESSAGE_CONSTRAINTS = "Ages must be smaller than 200.";
```

File: src/main/java/seedu/dengue/model/person/Age.java [hunk 7]
```diff
@@ -9,30 +9,30 @@
...
+            + " and smaller than 200.";
```
Comment (valerietanhx) [inline src/main/java/seedu/dengue/model/person/Age.java:13]:
```suggestion
```

File: docs/tutorials/AddRemark.md [hunk 8]
```diff
@@ -345,7 +345,7 @@ save it with `Model#setPerson()`.
 
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = new Person(
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getDate(),
```
Comment (valerietanhx) [inline docs/tutorials/AddRemark.md:348]:
```suggestion
                personToEdit.getName(), personToEdit.getPostal(), personToEdit.getDate(),
```

File: docs/UserGuide.md [hunk 9]
```diff
@@ -104,7 +106,7 @@ Format: `edit INDEX [n/NAME] [p/PHONE] [e/DATE] [a/AGE] [t/TAG]…​`
     specifying any tags after it.
 
 Examples:
-*  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and date age of the 1st person to be `91234567` and `johndoe@example.com` respectively.
+*  `edit 1 p/912345 d/2001-01-01` Edits the POSTAL number and date age of the 1st person to be `S912345` and `2001-01-01` respectively.
```
Comment (valerietanhx) [inline docs/UserGuide.md]:
```suggestion
*  `edit 1 p/912345 d/2001-01-01` Edits the postal code and date of the 1st person to be `S912345` and `2001-01-01` respectively.
```

**Review by valerietanhx — CHANGES_REQUESTED** [review]
Added a couple of suggestions, otherwise LGTM! Some comments can be handled in a separate PR instead :)

File: docs/UserGuide.md (hunk 1 above)
Comment (sumhungyee) [reply docs/UserGuide.md:30]: noted!

File: docs/diagrams/BetterModelClassDiagram.puml (hunk 2 above)
Comment (sumhungyee) [reply docs/diagrams/BetterModelClassDiagram.puml]: ok, next time

**Pull request merged**